src/main/java/baritone/process/elytra/ElytraBehavior.java

```diff
@@ -1262,6 +1280,7 @@ private static Vec3 step(final Vec3 motion, final Vec3 lookDirection, final floa
         return new Vec3(motionX, motionY, motionZ);
     }
 
+    // any call to this must be done with the lock held
```

Comment:
Can you maybe apply this comment to all methods which require synchronization and specify that the readlock is required?

Reply:
This is the only function run on the main thread that requires synchronization but i added a comment to solveAngles.

Reply:
As far as I can tell we have
```
passable called by simulate and pathFindAroundObstacles
simulate called by solvePitch called by the other solvePitch called by solveAngles
solveAngles called by onPostTick with synchronisation and by tick (and tick is public)
tick called by ElytraProcess.onTick with synchronization on the main thread
pathFindAroundObstacles called by PathManager.tick called by onTick0 called by onTick with synchronization on the main thread
```
At least `tick` needing synchronization should be documented.